crates/router/src/connector/prophetpay/transformers.rs

```diff
@@ -150,84 +373,136 @@ impl<F, T>
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+
+pub struct ProphetpayRefundResponse {
+    pub response_text: RefundStatus,
+    #[serde(rename = "transactionID")]
+    pub transaction_id: String,
 }
 
-impl TryFrom<types::RefundsResponseRouterData<api::Execute, RefundResponse>>
+impl TryFrom<types::RefundsResponseRouterData<api::Execute, ProphetpayRefundResponse>>
     for types::RefundsRouterData<api::Execute>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: types::RefundsResponseRouterData<api::Execute, RefundResponse>,
+        item: types::RefundsResponseRouterData<api::Execute, ProphetpayRefundResponse>,
     ) -> Result<Self, Self::Error> {
         Ok(Self {
             response: Ok(types::RefundsResponseData {
-                connector_refund_id: item.response.id.to_string(),
-                refund_status: enums::RefundStatus::from(item.response.status),
+                connector_refund_id: item.response.transaction_id.to_string(),
+                refund_status: enums::RefundStatus::from(item.response.response_text),
             }),
             ..item.data
         })
     }
 }
 
-impl TryFrom<types::RefundsResponseRouterData<api::RSync, RefundResponse>>
+#[derive(Debug, Clone, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ProphetpayRefundSyncRequest {
+    ref_info: String,
+    profile: Secret<String>,
+    action_type: i8,
+}
+
+impl TryFrom<&types::RefundSyncRouterData> for ProphetpayRefundSyncRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::RefundSyncRouterData) -> Result<Self, Self::Error> {
+        let auth_data = ProphetpayAuthType::try_from(&item.connector_auth_type)?;
+        Ok(Self {
+            ref_info: item.attempt_id.to_owned(),
+            profile: auth_data.profile_id,
+            action_type: get_action_type(ProphetpayActionType::Inquiry),
+        })
+    }
+}
+
+impl TryFrom<types::RefundsResponseRouterData<api::RSync, ProphetpayRefundResponse>>
     for types::RefundsRouterData<api::RSync>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: types::RefundsResponseRouterData<api::RSync, RefundResponse>,
+        item: types::RefundsResponseRouterData<api::RSync, ProphetpayRefundResponse>,
     ) -> Result<Self, Self::Error> {
         Ok(Self {
             response: Ok(types::RefundsResponseData {
-                connector_refund_id: item.response.id.to_string(),
-                refund_status: enums::RefundStatus::from(item.response.status),
+                connector_refund_id: item.response.transaction_id.to_string(),
+                refund_status: enums::RefundStatus::from(item.response.response_text),
             }),
             ..item.data
         })
     }
 }
 
 #[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
 pub struct ProphetpayErrorResponse {
```

Comment:
Add comment for ErrorResponse struct saying "Error Response body is yet to be confirmed with the connector"